src/pages/settings/Wallet/InternationalDepositAccount/utils.ts

```diff
@@ -0,0 +1,150 @@
+            acc[CONST.CORPAY_FIELDS.STEPS_NAME.BANK_INFORMATION][field.id] = field;
+        } else {
+            acc[CONST.CORPAY_FIELDS.STEPS_NAME.BANK_ACCOUNT_DETAILS] = acc[CONST.CORPAY_FIELDS.STEPS_NAME.BANK_ACCOUNT_DETAILS] ?? {};
+            acc[CONST.CORPAY_FIELDS.STEPS_NAME.BANK_ACCOUNT_DETAILS][field.id] = field;
+        }
+        return acc;
+    }, {} as Record<ValueOf<typeof CONST.CORPAY_FIELDS.STEPS_NAME>, CorpayFieldsMap>);
+}
+
+function getLatestCreatedBankAccount(bankAccountList: OnyxEntry<BankAccountList>): BankAccount | undefined {
+    return lodashSortBy(Object.values(bankAccountList ?? {}), 'accountData.created').pop();
+}
+
+function getSubstepValues(
+    privatePersonalDetails: OnyxEntry<PrivatePersonalDetails>,
+    corpayFields: OnyxEntry<CorpayFields>,
+    bankAccountList: OnyxEntry<BankAccountList>,
+    internationalBankAccountDraft: OnyxEntry<InternationalBankAccountForm>,
+    country: OnyxEntry<string>,
+    fieldsMap: Record<ValueOf<typeof CONST.CORPAY_FIELDS.STEPS_NAME>, CorpayFieldsMap>,
+): InternationalBankAccountForm {
+    const address = PersonalDetailsUtils.getCurrentAddress(privatePersonalDetails);
+    const personalDetailsFieldMap = fieldsMap[CONST.CORPAY_FIELDS.STEPS_NAME.ACCOUNT_HOLDER_INFORMATION];
+    const {street} = address ?? {};
+    const [street1, street2] = street ? street.split('\n') : [undefined, undefined];
+    const firstName = privatePersonalDetails?.legalFirstName ?? '';
+    const lastName = privatePersonalDetails?.legalLastName ?? '';
+    const fullName = `${firstName} ${lastName}`.trim() ? `${firstName} ${lastName}`.trim() : undefined;
+    const latestBankAccount = getLatestCreatedBankAccount(bankAccountList);
+    return {
+        ...internationalBankAccountDraft,
+        bankCountry: internationalBankAccountDraft?.bankCountry ?? corpayFields?.bankCountry ?? address?.country ?? latestBankAccount?.bankCountry ?? country ?? '',
+        bankCurrency: internationalBankAccountDraft?.bankCurrency ?? corpayFields?.bankCurrency,
+        accountHolderName: !isEmptyObject(personalDetailsFieldMap?.accountHolderName) ? internationalBankAccountDraft?.accountHolderName ?? fullName : undefined,
+        accountHolderAddress1: !isEmptyObject(personalDetailsFieldMap?.accountHolderAddress1) ? internationalBankAccountDraft?.accountHolderAddress1 ?? street1 : undefined,
+        accountHolderAddress2: !isEmptyObject(personalDetailsFieldMap?.accountHolderAddress2) ? internationalBankAccountDraft?.accountHolderAddress2 ?? street2 : undefined,
+        accountHolderCity: !isEmptyObject(personalDetailsFieldMap?.accountHolderCity) ? internationalBankAccountDraft?.accountHolderCity ?? address?.city : undefined,
+        accountHolderCountry: !isEmptyObject(personalDetailsFieldMap?.accountHolderCountry)
+            ? internationalBankAccountDraft?.accountHolderCountry ?? corpayFields?.bankCountry ?? address?.country ?? latestBankAccount?.bankCountry ?? country ?? ''
```

Comment:
@shubham1206agra This is accountHolderCountry.  Why do we use `corpayFields?.bankCountry`?

Reply:
From design doc, this field should use the data from private_personalDetails

<img width="687" alt="Screenshot 2024-12-09 at 16 34 34" src="https://github.com/user-attachments/assets/3575330d-60aa-4d2d-bd11-9534de93ae4a">